src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    public void applyOperation(Operation operation, String fruit,
+                               int quantity, Map<String, Integer> storage) {
+        operationHandlers.get(operation).apply(fruit, quantity, storage);

```

Comment:
The code does not handle the case when the 'operation' is not in the 'operationHandlers' Map. This could lead to a NullPointerException. It would be better to check if the 'operation' is in the Map before calling the 'apply' method.